samples/debugmalloc/README.md

```diff
@@ -0,0 +1,328 @@
+# The debug malloc sample
+
+- Introduction about debug malloc.
+- Demonstrate how to enable debug malloc through compiling options.
+- explanation and sample of debug malloc public apis.
+
+## Introduction
+
+Debug malloc is a tool that helps identifying memory leaks in enclaves. The
+main logic of debug malloc is as follows.
+
+- Whenever memory is allocated, debug malloc captures the callstack and
+associates it with the allocated memory. Additionally, debug malloc sets the
+contents of the allocated memory to zero.
+
+- Whenever memory is freed, it clears the contents to a fixed pattern to help
+detect access of freed memory. It also frees the associated callstack
+information.
+
+- When an enclave terminates, debug malloc prints information about leaks---
+those objects that have been allocated, but not freed.
+
+According to the user's developing experience, there are two major problems
+which the memory leaking detection can help to address:
+
+- A developer wants to track all memory allocations throughout the execution of an enclave
+  and to determine if any of these allocations were leaked.
+  We name it "global detection".
```

Comment:
```suggestion
  We call this "global detection".
```

Reply:
Also add text explaining whether this sample illustrates global detection or local detection or both.